tests/net.rs

```diff
@@ -154,6 +154,44 @@ fn hold_and_release_once_connected() -> turmoil::Result {
     sim.run()
 }
 
+#[test]
+fn accept_front_of_line_blocking() -> turmoil::Result {
+    let mut sim = Builder::new().build();
+
+    // We setup the simulation with hosts A, B, and C
+
+    sim.host("B", || async {
+        let listener = net::TcpListener::bind().await.unwrap();
+
+        while let Ok((_, peer)) = listener.accept().await {
+            debug!("peer {}", peer);
+        }
+    });
+
+    // Hold all traffic from A:B
+    sim.client("A", async {
+        hold("A", "B");
+
+        assert!(
+            timeout(Duration::from_secs(1), net::TcpStream::connect("B"))
+                .await
+                .is_err()
+        );
+
+        Ok(())
+    });
+
+    // C:B should succeed, and should not be blocked behind the A:B, which is
+    // not eligible for delivery
+    sim.client("C", async {
```

Comment:
Did this test fail before?

I'm not entirely sure how the scheduling works out here. What guarantees the hold (L173) happens before C:B syn on L187?

Reply:
It did fail due to deterministic host registration and execution. I'll add a waiter to ensure the order in case we change something.